awx/main/tasks.py

```diff
@@ -904,6 +904,23 @@ def build_execution_environment_params(self, instance):
             "container_options": ['--user=root'],
         }
 
+        if instance.execution_environment.credential:
+            with open('/tmp/auth.json', 'w') as authfile:
```

Comment:
@rebeccahhh @shanemcd should we randomize this filename?  If we don't, it seems like parallel jobs would step on eachother.

Reply:
Yup I think that's the plan and this is just a WIP

Reply:
Actually, is this something that belongs in the private data dir itself?

Reply:
Okey-doke, sorry if I'm being noisy 😄

Reply:
Probably not, because we don't want the user in the playbook to see the registry auth info.

Maybe it could go in the new wrapper directory we added?